gpio_test.go

```diff
@@ -0,0 +1,158 @@
+package gpio
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// valudate and extract test setup
+func getGpioSetup(t *testing.T) (gpiodev string, gpioin uint32, gpioout uint32) {
+	if len(os.Getenv("GPIO_DEV_PATH")) == 0 {
```

Comment:
```go
func getEnv(t *testing.T) (string, uint32, uint32) {
	dev := os.Getenv("GPIO_TEST_DEV")
	inStr := os.Getenv("GPIO_TEST_PIN_INPUT")
	outStr := os.Getenv("GPIO_TEST_PIN_OUTPUT")
	if dev == "" {
		t.SkipNow(...)
	}
```

- getGpioSetup -> getEnv, because everything here is about gpio already, this word adds noise to reading
- return variables declared but not used
- read environment once
- consistent GPIO_TEST_ prefix
- GPIO_DEV_PATH -> GPIO_TEST_DEV because PATH feels redundant here, up to you
- important to mark GPIO_TEST_PIN_INPUT/OUTPUT as they are used
- "idiomatic" to check empty string against without `len()`
- SkipNow, Fatalf to save on checking in tests below

Reply:
> return variables declared but not used

It is for IDE documentation purpose. Most IDEs will show the function prototype as hint

Reply:
Yeah I agree it's good documentation, especially when types are not so obvious like in `Atoi() string,error`
But since you declared return variables, might as well use them.
Updated example above.

Reply:
> important to mark GPIO_TEST_PIN_INPUT/OUTPUT as they are used

My reasoning behind the name is that it isn't "just" output but specifically "output looped to the other pin, so it is is immediately apparent from the variable name that you need to do more than just find first free GPIO pin.

Also there would be a conflict if say there would be tests added that use the dummy `gpio-mockup` driver as there is no way to loop pins to eachother in it (afaik)

Not that I *particularly* care about the environmental names.

I've changed variable names in code so they are indicative of the purpose the pin is used for at the given moment.